device/pcie/pci_device.hpp

```diff
@@ -34,84 +34,144 @@ struct PciDeviceInfo
     void write_regs(uint32_t byte_addr, uint32_t word_len, const void *data);
     void write_regs(volatile uint32_t *dest, const uint32_t *src, uint32_t word_len);
     void read_regs(uint32_t byte_addr, uint32_t word_len, void *data);
     void write_tlb_reg(uint32_t byte_addr, std::uint64_t value_lower, std::uint64_t value_upper, std::uint32_t tlb_cfg_reg_size);
 
-    void open_hugepage_per_host_mem_ch(uint32_t num_host_mem_channels);
     tt::umd::architecture_implementation* get_architecture_implementation() const { return architecture_implementation.get(); }
 
-    PciDeviceInfo info;
-
-    int device_id;  // N in /dev/tenstorrent/N
-    int logical_id; // TODO: does not belong in here
-    int device_fd = -1;
-
-    // PCIe device info
-    int numa_node;
-    std::uint32_t pcie_device_id;
-    std::uint32_t pcie_revision_id;
-
-    // BAR and regs mapping setup
-    std::vector<int> device_fd_per_host_ch;
+public:
+    // TODO: we can and should make all of these private.
     void *bar0_uc = nullptr;
-    std::size_t bar0_uc_size = 0;
-    std::size_t bar0_uc_offset = 0;
+    size_t bar0_uc_size = 0;
+    size_t bar0_uc_offset = 0;
 
     void *bar0_wc = nullptr;
-    std::size_t bar0_wc_size = 0;
+    size_t bar0_wc_size = 0;
 
     void *bar2_uc = nullptr;
-    std::size_t bar2_uc_size;
+    size_t bar2_uc_size;
 
     void *bar4_wc = nullptr;
-    std::uint64_t bar4_wc_size;
+    uint64_t bar4_wc_size;
 
+    // TODO: let's get rid of this unless we need to run UMD on WH systems with
+    // shrunk BAR0.  If we don't (and we shouldn't), then we can just use BAR0
+    // and simplify the code.
     void *system_reg_mapping = nullptr;
-    std::size_t system_reg_mapping_size;
-
-    // These two are currently not used.
-    void *system_reg_wc_mapping = nullptr;
-    std::size_t system_reg_wc_mapping_size;
-
-    std::uint32_t system_reg_start_offset;  // Registers >= this are system regs, use the mapping.
-    std::uint32_t system_reg_offset_adjust; // This is the offset of the first reg in the system reg mapping.
-
-    // int sysfs_config_fd = -1;    // not used
-    std::uint32_t read_checking_offset;
+    size_t system_reg_mapping_size;
+    uint32_t system_reg_start_offset;  // Registers >= this are system regs, use the mapping.
+    uint32_t system_reg_offset_adjust; // This is the offset of the first reg in the system reg mapping.
+
+    // TODO: this didn't survive a refactor.  It was part of some logic to
+    // detect a hung chip and reset it.  Detecting a hung chip is a good idea,
+    // but it should probably be a fatal error.  It is unclear if the recovery
+    // code was functional -- the "reset by sysfs" path certainly didn't look
+    // like it was.  Let's leave recovery up to the user + tt-smi for now, since
+    // UMD is in no position to actually be able to recover the hardware state
+    // on behalf of the application.
+    //
+    // But we should put back a if (hardware_is_hung()) TT_FATAL("it's dead");
```

Comment:
Yeah this is exactly what #144 does. I've removed auto_reset code from there.

Reply:
Ok cool - I will remove this comment when I rebase on main.